packages/ui/src/components/va-backtop/VaBacktop.vue

```diff
@@ -15,96 +15,89 @@
 </template>
 
 <script lang="ts">
-import { Options, mixins, prop, Vue } from 'vue-class-component'
+import { defineComponent, computed, PropType, ref, onMounted, onBeforeMount, Ref } from 'vue'
 
 import VaButton from '../va-button'
 
-class Props {
-  target = prop<Element | string | undefined>({
-    type: [Object, String],
-    default: () => window,
-  })
-
-  visibilityHeight = prop<number>({ type: Number, default: 300 })
-  speed = prop<number>({ type: Number, default: 50 })
-  verticalOffset = prop<string>({ type: String, default: '1rem' })
-  horizontalOffset = prop<string>({ type: String, default: '1rem' })
-  color = prop<string>({ type: String, default: '' })
-  horizontalPosition = prop<string>({
-    type: String,
-    default: 'right',
-    validator: (value: string) => {
-      return ['right', 'left'].includes(value)
+export default defineComponent({
+  name: 'VaBacktop',
+  components: { VaButton },
+  props: {
+    target: {
+      type: [Object, String] as PropType<Element | string>,
+      default: () => document.body,
     },
-  })
 
-  verticalPosition = prop<string>({
-    type: String,
-    default: 'bottom',
-    validator: (value: string) => {
-      return ['bottom', 'top'].includes(value)
+    visibilityHeight: { type: Number as PropType<number>, default: 300 },
+    speed: { type: Number as PropType<number>, default: 50 },
+    verticalOffset: { type: String as PropType<string>, default: '1rem' },
+    horizontalOffset: { type: String as PropType<string>, default: '1rem' },
+    color: { type: String as PropType<string>, default: '' },
+    horizontalPosition: {
+      type: String as PropType<'right' | 'left'>,
+      default: 'right',
+      validator: (value: string) => ['right', 'left'].includes(value),
     },
-  })
-}
 
-const PropsMixin = Vue.with(Props)
+    verticalPosition: {
+      type: String as PropType<'bottom' | 'top'>,
+      default: 'bottom',
+      validator: (value: string) => ['bottom', 'top'].includes(value),
+    },
+  },
+  setup (props) {
+    const visible = ref(false)
 
-@Options({
-  name: 'VaBacktop',
-  components: { VaButton },
-})
-export default class VaBacktop extends mixins(
-  PropsMixin,
-) {
-  visible = false
-  scrolled = false
-  interval = 0
-
-  get computedStyle () {
-    return {
-      [this.verticalPosition]: this.verticalOffset,
-      [this.horizontalPosition]: this.horizontalOffset,
+    const computedStyle = computed(() => ({
+      [props.verticalPosition]: props.verticalOffset,
+      [props.horizontalPosition]: props.horizontalOffset,
+    }))
+
+    const targetElement: Ref<Element> = ref(document.body)
+    const getTargetElement = () => typeof props.target === 'string'
+      ? document.querySelector(props.target) || document.body
+      : props.target
```

Comment:
Maybe we can do this:
`const targetElement = computed<Element>(() => typeof props.target === 'string' ? document.querySelector(props.target) || document.body : props.target)`
And maybe instead of `document.body` we have to use **parent of component**?

Reply:
> Maybe we can do this:
That's what I tried initially but the target is not always mounted when the computed is running which results in target being null.
>And maybe instead of document.body we have to use parent of component?
I don't mind but the docs says default target is body.